frontends/web/src/components/dialog/dialog.tsx

```diff
@@ -32,228 +34,195 @@ interface Props {
     open: boolean;
 }
 
-interface State {
-  currentTab: number;
-  renderDialog: boolean;
-}
-
-class Dialog extends Component<Props, State> {
-  private overlay = createRef<HTMLDivElement>();
-  private modal = createRef<HTMLDivElement>();
-  private modalContent = createRef<HTMLDivElement>();
-  private focusableChildren!: NodeListOf<HTMLElement>;
-  private timerId: any;
-
-  public state: State = {
-    currentTab: 0,
-    renderDialog: false,
-  };
-
-  public componentDidMount(): void {
-    if (this.props.open) {
-      this.activate();
+export const Dialog = ({
+  title,
+  small,
+  medium,
+  large,
+  slim,
+  centered,
+  disableEscape,
+  onClose,
+  disabledClose,
+  children,
+  open,
+}: TProps) => {
+  const [currentTab, setCurrentTab] = useState<number>(0);
+  const [renderDialog, setRenderDialog] = useState<boolean>(false);
+  const overlayRef = useRef<HTMLDivElement>(null);
+  const modalRef = useRef<HTMLDivElement>(null);
+  const modalContentRef = useRef<HTMLDivElement>(null);
+  const timerIdRef = useRef<any>(null);
```

Comment:
in send.tsx we use `ReturnType<typeof setTimeout>` instead of `any`.

Reply:
fixed